### PR TITLE
Features/update nightwatch and use deafault config

### PR DESCRIPTION
**Pull request by gravityvi**
### Changes
- updated scripts and dependencies to support Nightwatch default config
- ~~added Nightwatch default config.~~
- ~~updated the config to support all environments (local, parallel, etc)~~ --> Will work with nightwatch default config 
- updated the env variables as per Nightwatch config.
- removed separate files for configs
- updated README
- removed deprecated methods in `local.runner`

### CODE REVIEW

**Review by vaibhavsingh97 — APPROVED** [review]

File: nightwatch.conf.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+    'browserstack.local': {
+      extends: 'browserstack',
+      desiredCapabilities: {
+        'browserstack.local': 'true'
+      }
+    },
+
+    'browserstack.local_chrome': {
+      extends: 'browserstack',
```
Comment (beatfactor) [inline nightwatch.conf.js:148]: does it work to extend `browserstack.local` here?

File: scripts/local.runner.js [hunk 2]
```diff
@@ -5,24 +5,23 @@ var browserstack = require('browserstack-local');
 var bs_local;
 
 try {
-
-  process.mainModule.filename = "./node_modules/nightwatch/bin/nightwatch"
+  require.main.filename = './node_modules/nightwatch/bin/nightwatch';
```
Comment (beatfactor) [inline scripts/local.runner.js:8]: should be './node_modules/.bin/nightwatch'

File: scripts/local.runner.js [hunk 3]
```diff
@@ -5,24 +5,23 @@ var browserstack = require('browserstack-local');
 var bs_local;
```
Comment (beatfactor) [inline scripts/local.runner.js:5]: maybe use let and const here

File: scripts/local.runner.js [hunk 4]
```diff
@@ -5,24 +5,23 @@ var browserstack = require('browserstack-local');
...
+        .setup(null, function () {
           // Code to stop browserstack local after end of parallel test
-          bs_local.stop(function(){});
+          bs_local.stop(function () {});
         })
-        .runTests(function(){
+        .runTests(function () {
           // Code to stop browserstack local after end of single test
-          bs_local.stop(function(){});
+          bs_local.stop(function () {});
```
Comment (beatfactor) [inline scripts/local.runner.js:24]: runTests() is returning a promise. it should be handled like in https://github.com/nightwatchjs/nightwatch/blob/main/bin/runner.js#L18

File: package.json [hunk 5]
```diff
@@ -5,19 +5,19 @@
...
-    "single": "./node_modules/.bin/nightwatch -c conf/single.conf.js",
-    "local": "node scripts/local.runner.js -c conf/local.conf.js",
-    "parallel": "./node_modules/.bin/nightwatch -c conf/parallel.conf.js -e chrome,firefox,safari,ie",
-    "parallel_local": "node scripts/local.runner.js -c conf/parallel_local.conf.js -e chrome,firefox,safari",
-    "suite": "./node_modules/.bin/nightwatch -c conf/suite.conf.js"
+    "single": "nightwatch --group single --env browserstack",
+    "local": "node scripts/local.runner.js --group local --env browserstack.local",
+    "parallel": "nightwatch --group single --env browserstack.chrome,browserstack.ie,browserstack.safari,browserstack.firefox",
+    "parallel_local": "node scripts/local.runner.js --group local --env browserstack.local_firefox,browserstack.local_safari,browserstack.local_chrome",
+    "suite": "nightwatch --group suite --parallel --env browserstack.chrome --verbose"
```
Comment (beatfactor) [inline package.json:12]: maybe remove the --verbose?

File: nightwatch.conf.js (hunk 1 above)
Comment (gravityvi) [reply nightwatch.conf.js:148]: Nope, multiple `extends` does not seem to work

File: scripts/local.runner.js (hunk 3 above)
Comment (gravityvi) [reply scripts/local.runner.js:5]: sure, will make the change 👍🏻

File: scripts/local.runner.js (hunk 4 above)
Comment (gravityvi) [reply scripts/local.runner.js:24]: Will make the change 👍🏻

File: package.json (hunk 5 above)
Comment (gravityvi) [reply package.json:12]: yes, will make the change

**Review by beatfactor — APPROVED** [review]

**Pull request merged**